src/wp-includes/class-wp-theme-json.php

```diff
@@ -764,6 +764,18 @@ private function get_block_classes( $style_nodes ) {
 				}
 			}
 
+			/*
+			 * Reset default browser margin on the root body element.
+			 * This is set on the root selector **before** generating the ruleset
+			 * from the `theme.json`. This is to ensure that if the `theme.json` declares
+			 * `margin` in its `spacing` declaration for the `body` element then these
+			 * user-generated values take precedence in the CSS cascade.
+			 * @link https://github.com/WordPress/gutenberg/issues/36147.
+			 */
+			if ( self::ROOT_BLOCK_SELECTOR === $selector ) {
+				$block_rules .= "body { margin: 0; }\n";
```

Comment:
@getdave There are 6 failures in the PHPUnit tests from the changes in this PR. https://github.com/WordPress/wordpress-develop/runs/4396256568?check_suite_focus=true 

Before:
```css
body { margin: 0; }.wp-site-blocks > .alignleft {
```

Actual (after this change):
```css
body { margin: 0; }
.wp-site-blocks > .alignleft {
```

The failures are valid as the placement of the body margin changed to be on its own line `"\n"`. Is the newline needed? If yes, the tests need updating.